### PR TITLE
Refactor the folder structure to organize recipes by topic

### PR DESCRIPTION
**Pull request by subramen**
# What does this PR do?
The PR refactors recipes in the current repo to be organized by topic. Contents within `demo_apps/` and `examples/` are moved into `recipes/`. 

With the new folders, the PR also creates new folder-level READMEs to provide information about the recipes in the same folder.  The main README has also been minimized and the existing content has been refactored into the relevant sections of the repo.

These changes are part of the broader effort to improve the developer experience of using this repository.

### CODE REVIEW

File: recipes/README.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
+This folder contains examples organized by topic:
+
+| Subfolder | Description |
+|---|---|
```
Comment (HamidShojanazeri) [inline recipes/README.md:4]: perhaps similar ordering as above.

File: recipes/quickstart/Running_Llama2_Anywhere/Running_Llama_on_HF_transformers.ipynb [hunk 2]
```diff
@@ -0,0 +1,304 @@
+{
```
Comment (HamidShojanazeri) [inline recipes/quickstart/Running_Llama2_Anywhere/Running_Llama_on_HF_transformers.ipynb:1]: is this new? can it be a google colab?

File: recipes/responsible_ai/README.md [hunk 3]
```diff
@@ -0,0 +1,11 @@
+# Llama Guard
+
+Llama Guard is a new experimental model that provides input and output guardrails for LLM deployments. For more details, please visit the main [repository](https://github.com/facebookresearch/PurpleLlama/tree/main/Llama-Guard).
```
Comment (HamidShojanazeri) [inline recipes/responsible_ai/README.md:3]: is it experimental model?

**Review by HamidShojanazeri — COMMENTED** [review]
Thanks @subramen for the PR!  added few initial comments.

File: recipes/quickstart/Running_Llama2_Anywhere/Running_Llama_on_HF_transformers.ipynb (hunk 2 above)
Comment (subramen) [reply recipes/quickstart/Running_Llama2_Anywhere/Running_Llama_on_HF_transformers.ipynb:1]: I think that would be a follow up after this PR cc @fbnav

File: recipes/responsible_ai/README.md (hunk 3 above)
Comment (subramen) [reply recipes/responsible_ai/README.md:3]: Updated to reflect current messaging

**Comment by HamidShojanazeri** [comment]
I am also thinking maybe we could maintain a table for main features to make it easier for users to track?

**Pull request closed without merging**